tests/users/test_data.py

```diff
@@ -13,49 +15,75 @@
 from virtool.fake.next import DataFaker
 from virtool.mongo.core import Mongo
 from virtool.mongo.utils import get_one_field
+from virtool.pg.utils import get_row_by_id
 from virtool.users.db import B2CUserAttributes
 from virtool.users.mongo import validate_credentials
 from virtool.users.oas import UpdateUserRequest
+from virtool.users.pg import SQLUser, user_group_associations
 
 _last_password_change_matcher = path_type(
     {"last_password_change": (datetime.datetime,)}
 )
 
 
+@pytest.fixture(params=[0, 1, 2])
+def groups_param(request):
+    return request.param
+
+
+@pytest.fixture
+async def groups(groups_param, fake2):
+    groups = []
+    for _ in range(groups_param):
+        groups.append(await fake2.groups.create())
+    return groups
+
+
```

Comment:
What's the advantage over `@parameterise`?

Reply:
misunderstood how I had to get the 0,1, or 2 groups. Condensed into 1 function now. but we need this to simulate the actual groups vs just group id's.

Reply:
Is there a blocker on doing it in the main test function? something like:
`groups = await asyncio.gather([fake2.groups.create() for group in range(groups_count) ])`